src/AWS.Deploy.Common/DeploymentManifest/DeploymentManifestEngine.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using AWS.Deploy.Common.IO;
+using Newtonsoft.Json;
+
+namespace AWS.Deploy.Common.DeploymentManifest
+{
+    /// <summary>
+    /// This class contains the helper methods to update the deployment manifest file
+    /// that keeps track of the save CDK deployment projects.
+    /// </summary>
+    public class DeploymentManifestEngine
+    {
+        private readonly IDirectoryManager _directoryManager;
+        private readonly IFileManager _fileManager;
+        private readonly string _targetApplicationFullPath;
+        private readonly string _saveCdkDirectoryFullPath;
+
+        public DeploymentManifestEngine(string targetApplicationFullPath, string saveCdkDirectoryFullPath)
+        {
+            _directoryManager = new DirectoryManager();
+            _fileManager = new FileManager();
```

Comment:
Don't initialize in the constructor, use them as constructor dependencies. Both `IFileManager` and `IDirectoryManager` are available in DI 

https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/Extensions/CustomServiceCollectionExtension.cs#L40

Reply:
Fixed